server/src/main/java/com/thoughtworks/go/server/dashboard/GoDashboardCurrentStateLoader.java

```diff
@@ -144,10 +145,19 @@ private PipelineInstanceModels loadHistoryForPipelines(List<String> pipelineName
 
     public GoDashboardPipeline pipelineFor(PipelineConfig pipelineConfig, PipelineConfigs groupConfig) {
         List<String> pipelineNames = CaseInsensitiveString.toStringList(Collections.singletonList(pipelineConfig.getName()));
-        PipelineInstanceModels historyForDashboard = loadHistoryForPipelines(pipelineNames);
+        PipelineInstanceModels pipelineHistoryForDashboard = loadHistoryForPipelines(pipelineNames);
+
+        List<PipelineInstanceModel> allPipelinesExceptPipelineConfigInContext = historyForDashboard.stream().filter(new Predicate<PipelineInstanceModel>() {
+            @Override
+            public boolean test(PipelineInstanceModel pipelineInstanceModel) {
+                return !pipelineInstanceModel.getName().equalsIgnoreCase(pipelineConfig.name().toString());
+            }
+        }).collect(toList());
+        historyForDashboard = createPipelineInstanceModels(allPipelinesExceptPipelineConfigInContext);
+        historyForDashboard.addAll(pipelineHistoryForDashboard);
```

Comment:
LGTM. Extract into some `mergeWithHistoryForDashboard()` method (or similar name) to clarify intent?